charts/prefect-worker/templates/_helpers.tpl

```diff
@@ -58,38 +58,25 @@ Require Self-managed Cloud API URL
 Require Prefect Server API URL
 */}}
 {{- define "server.requiredConfig.apiUrl" -}}
-{{- if eq .Values.worker.apiConfig "server" }}
-    {{- required "The Prefect Server API URL is required (worker.serverApiConfig.apiUrl)" .Values.worker.serverApiConfig.apiUrl -}}
+{{- if eq .Values.worker.apiConfig "selfHostedServer" }}
+    {{- required "The Prefect Server API URL is required (worker.selfHostedServerApiConfig.apiUrl)" .Values.worker.selfHostedServerApiConfig.apiUrl -}}
 {{- end -}}
 {{- end -}}
 
 {{/*
   worker.apiUrl:
-    Define API URL for cloud or server worker install
+    Define the API URL for the worker based on the API config
 */}}
 {{- define "worker.apiUrl" -}}
 {{- if eq .Values.worker.apiConfig "cloud" }}
     {{- printf "%s/accounts/%s/workspaces/%s" .Values.worker.cloudApiConfig.cloudUrl (include "cloud.requiredConfig.accountId" .) (include "cloud.requiredConfig.workspaceId" .) | quote }}
-{{- else if eq .Values.worker.apiConfig "selfManaged" }}
+{{- else if eq .Values.worker.apiConfig "selfManagedCloud" }}
     {{- printf "%s/accounts/%s/workspaces/%s" (include "selfManaged.requiredConfig.apiUrl" .) (include "selfManaged.requiredConfig.accountId" .) (include "selfManaged.requiredConfig.workspaceId" .) | quote }}
-{{- else }}
+{{- else if eq .Values.worker.apiConfig "selfHostedServer" }}
     {{- include "server.requiredConfig.apiUrl" . | quote }}
 {{- end }}
 {{- end }}
 
-{{/*
```

Comment:
This is related to my note [here](https://github.com/PrefectHQ/prefect-helm/pull/453/files#r1966018760)